refinery/ui/source/js/tool-launch/services/tool-launch-service.js

```diff
@@ -258,5 +259,14 @@
       var tool = toolsService.save(launchConfig);
       return tool.$promise;
     }
+
+    /**
+     * @name userIsAnonymous
+     * @desc check if the user trying to launch a tool is anonymous or not
+     * @memberOf refineryToolLaunch.ToolLaunchButtonCtrl
+     */
+    function userIsAnonymous () {
+      return $window.user_id === undefined;
```

Comment:
At some point (soon), the front end (within angular) should know about who's logged in and we should avoid the $window object. (Technical debt?) Is the $window.user_id property always defined and actually set to undefined for anon user? If the user_id is not a property of the $window, an error is thrown.

Reply:
I poked around and found the `auth` service in commons so I'm using that now